.github/workflows/ci.yml

```diff
@@ -712,3 +725,144 @@ jobs:
+
+          # Validate that this commit is that latest commit for the branch using GitHub API
+          # We need to check this in case re-rerunning an old job and there have been new commits since
+          if [[ $DO_RELEASE == "1" ]]; then
+            # https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28
+            AMP='&'
+            RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+            -X GET https://api.github.com/repos/${GITHUB_REPOSITORY}/commits?sha=${GITHUB_REF_NAME}${AMP}per_page=1 \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            )
+            if [[ $RESP_CODE != "200" ]]; then
+              echo "Unable to read list of commits - HTTP response code was $RESP_CODE"
+              exit 1
+            fi
+            LATEST_REMOTE_SHA=$(jq -r '.[0].sha' __response.json)
+            echo "LATEST_REMOTE_SHA is $LATEST_REMOTE_SHA"
+            echo "LATEST_LOCAL_SHA is $LATEST_LOCAL_SHA"
+            if [[ $LATEST_REMOTE_SHA != $LATEST_LOCAL_SHA ]]; then
+              echo "Not patch releasing because latest remote sha is not equal to latest local sha"
+              DO_RELEASE=0
+            fi
+            # Also validate the sha matches GITHUB_SHA, which is what gha-tag-release will use
+            if [[ $GITHUB_SHA != $LATEST_LOCAL_SHA ]]; then
+              echo "Not patch releasing because GITHUB_SHA is not equal to latest local sha"
+              DO_RELEASE=0
+            fi
+          fi
+
+          # Check is there is an existing tag on the branch using GitHub API
+          # Note cannot use local `git tag` because actions/checkout by default will not checkout tags
+          # and you need to checkout full history in order to get them
+          LATEST_TAG=""
+          NEXT_TAG=""
+          if [[ $DO_RELEASE == "1" ]]; then
+            # https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28
+            RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+            -X GET https://api.github.com/repos/${GITHUB_REPOSITORY}/tags?per_page=100 \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            )
+            if [[ $RESP_CODE != "200" ]]; then
+              echo "Unable to read list of tags - HTTP response code was $RESP_CODE"
+              exit 1
+            fi
+            # Get the latest tag
+            # Escape for regex e.g. 2.0 => 2\.0
+            ESCAPED_REF_NAME="${GITHUB_REF_NAME/./"\."}"
+            LATEST_TAG=$(jq -r '.[].name' __response.json | grep -E ^$ESCAPED_REF_NAME\.[0-9]+$ | sort -V -r | head -n 1)
+            echo "LATEST_TAG is $LATEST_TAG"
+            echo "latest_tag=$LATEST_TAG" >> $GITHUB_OUTPUT
+            if ! [[ $LATEST_TAG =~ ([0-9]+)\.([0-9]+)\.([0-9]+) ]]; then
```

Comment:
I don't see the point of this condition, since we're already using REGEX to get the latest tag name anyway? Why don't we just require the latest tag to be a patch tag in the first set of regex?

Reply:
$LATEST_TAG could be blank (or something else, I don't know what it would be returned) if there are zero tags that match